src/main/java/com/testmonitor/actions/TestCaseFolders.java

```diff
@@ -200,6 +202,39 @@ public TestCaseFolder clone(TestCaseFolder source, String name) throws IOExcepti
         return this.clone(source, name, true, true);
     }
 
+    /**
+     * Find a test case folder using the provided name or create a new one.
+     *
+     * @param name The name of the test case folder
+     *
+     * @return A test case folder matching the name or a new test case folder
+     */
+    public TestCaseFolder findOrCreate(String name) throws IOException, URISyntaxException {
+        ArrayList<TestCaseFolder> testCaseFolders = this.findByName(name);
+
+        if (testCaseFolders.size() > 0) {
+            return testCaseFolders.get(0);
+        }
+
+        return this.create(name);
+    }
+
+    /**
+     * Find a test case folder using the provided name and test folder.
+     *
+     * @param name The name of the test case folder
+     *
+     * @return Test case folders matching the provided name.
+     */
+    public ArrayList<TestCaseFolder> findByName(String name) throws IOException, URISyntaxException {
```

Comment:
Only root level?

This method probably needs a parent_id as well.. or overload it.

Reply:
Ok, I added support for trees.